mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -0,0 +1,660 @@
+    }
+
+    fn get_receiver_trade_fee(&self, stage: FeeApproxStage) -> TradePreimageFut<TradeFee> { todo!() }
+
+    async fn get_fee_to_send_taker_fee(
+        &self,
+        dex_fee_amount: BigDecimal,
+        stage: FeeApproxStage,
+    ) -> TradePreimageResult<TradeFee> {
+        todo!()
+    }
+
+    fn required_confirmations(&self) -> u64 { todo!() }
+
+    fn requires_notarization(&self) -> bool { todo!() }
+
+    fn set_required_confirmations(&self, confirmations: u64) { todo!() }
+
+    fn set_requires_notarization(&self, requires_nota: bool) { todo!() }
+
+    fn swap_contract_address(&self) -> Option<BytesJson> { todo!() }
+
+    fn mature_confirmations(&self) -> Option<u32> { None }
+
+    fn coin_protocol_info(&self) -> Vec<u8> { Vec::new() }
+
+    fn is_coin_protocol_supported(&self, info: &Option<Vec<u8>>) -> bool { true }
+}
+
+#[allow(unused_variables)]
+impl MarketCoinOps for TendermintCoin {
+    fn ticker(&self) -> &str { &self.ticker }
+
+    fn my_address(&self) -> Result<String, String> { Ok(self.account_id.to_string()) }
+
+    fn get_public_key(&self) -> Result<String, MmError<UnexpectedDerivationMethod>> { todo!() }
+
+    fn sign_message_hash(&self, _message: &str) -> Option<[u8; 32]> { todo!() }
+
+    fn sign_message(&self, _message: &str) -> SignatureResult<String> { todo!() }
+
+    fn verify_message(&self, _signature: &str, _message: &str, _address: &str) -> VerificationResult<bool> { todo!() }
+
+    fn my_balance(&self) -> BalanceFut<CoinBalance> {
+        let coin = self.clone();
+        let fut = async move {
+            let balance_denom = coin.my_balance_denom().await?;
+            Ok(CoinBalance {
+                spendable: big_decimal_from_sat_unsigned(balance_denom, coin.decimals),
+                unspendable: BigDecimal::default(),
+            })
+        };
+        Box::new(fut.boxed().compat())
+    }
+
+    fn base_coin_balance(&self) -> BalanceFut<BigDecimal> {
+        Box::new(self.my_balance().map(|coin_balance| coin_balance.spendable))
+    }
+
+    fn platform_ticker(&self) -> &str { &self.ticker }
+
+    fn send_raw_tx(&self, tx: &str) -> Box<dyn Future<Item = String, Error = String> + Send> {
+        let tx_bytes = try_fus!(hex::decode(tx));
+        let tx = try_fus!(Raw::from_bytes(&tx_bytes));
+        let coin = self.clone();
+        let fut = async move {
+            let broadcast_res = try_s!(coin.rpc_client.broadcast_tx_commit(tx_bytes.into()).await);
+            if !broadcast_res.check_tx.code.is_ok() {
+                return ERR!("Tx check failed {:?}", broadcast_res.check_tx);
+            }
+
+            if !broadcast_res.deliver_tx.code.is_ok() {
+                return ERR!("Tx deliver failed {:?}", broadcast_res.deliver_tx);
+            }
+            Ok(broadcast_res.hash.to_string())
+        };
+        Box::new(fut.boxed().compat())
+    }
+
+    fn send_raw_tx_bytes(&self, tx: &[u8]) -> Box<dyn Future<Item = String, Error = String> + Send> {
```

Comment:
Can we add a. common function for `send_raw_tx` and `send_raw_tx_bytes`?

Reply:
`send_raw_tx` can reuse `send_raw_tx_bytes`